sklearnex/decomposition/pca.py

```diff
@@ -156,32 +162,38 @@ def transform(self, X):
 
         def _onedal_transform(self, X, queue=None):
             check_is_fitted(self)
+            if sklearn_check_version("1.0"):
+                self._check_feature_names(X, reset=False)
             X = self._validate_data(
                 X,
                 dtype=[np.float64, np.float32],
                 reset=False,
             )
             self._validate_n_features_in_after_fitting(X)
-            if sklearn_check_version("1.0"):
-                self._check_feature_names(X, reset=False)
 
             return self._onedal_estimator.predict(X, queue=queue)
 
         def fit_transform(self, X, y=None):
-            U, S, Vt = self._fit(X)
-            if U is None:
+            if sklearn_check_version("1.5"):
+                U, S, Vt, X_fit, x_is_centered, xp = self._fit(X)
+            else:
+                U, S, Vt = self._fit(X)
+            if hasattr(self, "_onedal_estimator"):
                 # oneDAL PCA was fit
                 return self.transform(X)
-            else:
+            elif U is not None:
                 # Scikit-learn PCA was fit
                 U = U[:, : self.n_components_]
 
                 if self.whiten:
-                    U *= sqrt(X.shape[0] - 1)
+                    U *= sqrt(X_fit.shape[0] - 1)
                 else:
                     U *= S[: self.n_components_]
 
                 return U
+            else:

```

Comment:
The current default method of sklearnex PCA is "covariance_eigh" although we call it "full". We don't have to fallback to sklearn. The _is_solver_compatible method can be updated to return true if the method=="covariance_eigh".